File: 56. Merge Intervals/56.go
package leetcode

import(
	"sort"
)

func merge(intervals [][]int) [][]int {
	if len(intervals) < 2{
		return intervals
	}
    sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})
	res := [][]int{intervals[0]}
	for _, v := range intervals{
		if v[0]>res[len(res)-1][1]{
			res = append(res, v)
		}else{
			tmpmax := 0
			if res[len(res)-1][1]>v[1]{
				tmpmax = res[len(res)-1][1]
			}else{
				tmpmax = v[1]
			}
			res[len(res)-1] = []int{res[len(res)-1][0], tmpmax}
		}
	}
	return res
}

File: 56. Merge Intervals/_56.py
class Solution:
    def merge(self, intervals):
        if not intervals:
            return []
        intervals.sort(key = lambda x: x[0])
        res  = [intervals[0]]
        for i in intervals:
            if res[-1][-1]<i[0]:
                res.append(i)
            else:
                res[-1] = [res[-1][0], max(res[-1][-1], i[-1])]
        return res

    def merge1(self, intervals):
        out = []
        for i in sorted(intervals, key=lambda i: i.start):
            if out and i.start <= out[-1].end:
                out[-1].end = max(out[-1].end, i.end)
            else:
                out += i,
        return out